### PR TITLE
INTMDB-714: [CFN - PI] SearchIndex: Unable to cast (*string)(0xc0005f04b0) of type *string to map[string]interface{}

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ [INTMDB-714](https://jira.mongodb.org/browse/INTMDB-714)

Issue: The searchindex resource does not set correctly `Mappings.Fields`. Updating `Mappings.Fields` to be an array of strings with the format "FieldName:FieldType" as we do for AtlasCLI.

Link to any related issue(s):  #482 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update

## Manual QA performed:

- [x] cfn invoke for each of CRUDL/cfn test
- [x] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [x] Published to AWS private registry
- [x] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [x] Deleted stack to ensure resources are deleted
- [x] Created multiple resources in same stack
- [x] Validated in Atlas UI
- [x] Included screenshots

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

## Next Steps

- Release the CFN resource
- Fix the CDK resource
- Release the CDK resource

## Further comments

```bash
cfn test
================================================================================= 8 passed, 4 skipped, 12 deselected, 1 warning in 1160.25s (0:19:20) ===
```

**Manual testing:**

Screenshot 2023-04-03 at 18 17 26

Screenshot 2023-04-03 at 18 17 51

Screenshot 2023-04-03 at 18 21 19

### CODE REVIEW

File: cfn-resources/search-index/docs/apiatlasftsmappingsviewmanual.md [hunk 1]
```diff
@@ -34,11 +35,11 @@ _Update requires_: [No interruption](https://docs.aws.amazon.com/AWSCloudFormati
 
 #### Fields
 
-One or more field specifications for the Atlas Search index. Required if **mappings.dynamic** is omitted or set to **false**.
+One or more field specifications for the Atlas Search index. The element of the array must have the format fieldName:fieldType. Required if **mappings.dynamic** is omitted or set to **false**.
```
Comment (andreaangiolillo) [inline cfn-resources/search-index/docs/apiatlasftsmappingsviewmanual.md:38]: Updated the documentation

File: examples/search-index/searchIndexFields.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+AWSTemplateFormatVersion: "2010-09-09"
```
Comment (colm-quinn) [inline examples/search-index/searchIndexFields.yaml:1]: [q] Does the manual QA docs in the repo also guide regression testing to use this example also?

**Review by colm-quinn — APPROVED** [review]
LGTM, just a small question on testing

**Comment by colm-quinn** [comment]
Just looking at testing w/ github actions, the contract tests were skipped here. Is that intentional or we don't have testing support for search index yet?

**Comment by andreaangiolillo** [comment]
@colm-quinn Right now we have contract testing only for **projects** and **auditing**. 

Action Logic: 

- We first run `change-detection` to detect the files changed in the PR
- We run the contract testing of a resource `X`, only if the PR changes at least one file in its folder `cfn-resources/X`

GitHub shows this nice diagram to help understand how the action works [4599801907](https://github.com/mongodb/mongodbatlas-cloudformation-resources/actions/runs/4599801907?pr=486).

File: examples/search-index/searchIndexFields.yaml (hunk 2 above)
Comment (andreaangiolillo) [reply examples/search-index/searchIndexFields.yaml:1]: no, good catch. I will update the manual QA file in the follow-up PR for the CDK L1 constructor

**Pull request merged**